tests/v2/codecoverage/setuprancher/main.go

```diff
@@ -0,0 +1,511 @@
+	AdminPassword string   `json:"AdminPassword" yaml:"AdminPassword"`
+}
+
+// setup for code coverage testing and reporting
+func main() {
+	rancherConfig := new(rancher.Config)
+	config.LoadConfig(rancher.ConfigurationFileKey, rancherConfig)
+
+	kubeconfig, err := getRancherKubeconfig()
+	if err != nil {
+		logrus.Fatalf("error with getting kube config using corral: %v", err)
+	}
+
+	password, err := corral.GetCorralEnvVar(corralName, "bootstrap_password")
+	if err != nil {
+		logrus.Fatalf("error getting password %v", err)
+	}
+
+	// update deployment
+	err = updateRancherDeployment(kubeconfig)
+	if err != nil {
+		logrus.Fatalf("error updating rancher deployment: %v", err)
+	}
+
+	token, err := createAdminToken(password, rancherConfig)
+	if err != nil {
+		logrus.Fatalf("error with generating admin token: %v", err)
+	}
+
+	rancherConfig.AdminToken = token
+	config.UpdateConfig(rancher.ConfigurationFileKey, rancherConfig)
+	//provision clusters for test
+	session := session.NewSession()
+	clustersConfig := new(provisioning.Config)
+	config.LoadConfig(provisioning.ConfigurationFileKey, clustersConfig)
+	kubernetesVersions := clustersConfig.RKE1KubernetesVersions
+	cnis := clustersConfig.CNIs
+	nodesAndRoles := clustersConfig.NodesAndRolesRKE1
+
+	client, err := rancher.NewClient("", session)
+	if err != nil {
+		logrus.Fatalf("error creating admin client: %v", err)
+	}
+
+	err = postRancherInstall(client)
+	if err != nil {
+		logrus.Fatalf("error with admin user service account secret: %v", err)
+	}
+
+	enabled := true
+	var testuser = namegen.AppendRandomString("testuser-")
+	var testpassword = passwordgenerator.GenerateUserPassword("testpass-")
+	user := &management.User{
+		Username: testuser,
+		Password: testpassword,
+		Name:     testuser,
+		Enabled:  &enabled,
+	}
+
+	newUser, err := users.CreateUserWithRole(client, user, "user")
+	if err != nil {
+		logrus.Fatalf("error creating admin client: %v", err)
+	}
+
+	newUser.Password = user.Password
+
+	standardUserClient, err := client.AsUser(newUser)
+	if err != nil {
+		logrus.Fatalf("error creating standard user client: %v", err)
+	}
+
+	// create admin cluster
+	adminClusterNames, err := createTestCluster(client, client, 1, "admintestcluster", cnis[0], kubernetesVersions[0], nodesAndRoles)
+	if err != nil {
+		logrus.Fatalf("error creating admin user cluster: %v", err)
+	}
+
+	// create standard user clusters
+	standardClusterNames, err := createTestCluster(standardUserClient, client, 2, "standardtestcluster", cnis[0], kubernetesVersions[0], nodesAndRoles)
+	if err != nil {
+		logrus.Fatalf("error creating standard user clusters: %v", err)
+	}
+
+	//update userconfig
+	userClusterConfig := UserClusterConfig{}
+	userClusterConfig.Token = standardUserClient.Steve.Opts.TokenKey
+	userClusterConfig.Username = newUser.Name
+	userClusterConfig.AdminPassword = adminPassword
+	userClusterConfig.Clusters = standardClusterNames
+
+	rancherConfig.ClusterName = adminClusterNames[0]
```

Comment:
Shouldn't `ClusterName` be a slice instead of a string so we could set multiple clusters for admin like with userClusterConfig?

Reply:
As of right now that isn't the behavior of the framework. Maybe that is something we could implement in the future ,but how it is designed right now. We only need one cluster when testing. I'm indexing the slice because createTestCluster returns a slice

Reply:
If it's by design why do we need an slice?

Reply:
Because of the function `createTestCluster` on line 358

Reply:
My bad, what I tried to ask is why the design only considers 1 cluster for admin but there are cases where there are multiple clusters for other users (`userClusterConfig`). The Rancher admin can also create multiple clusters. I am probably confused as why this shouldn't behave as how Rancher does.

Like the case here with `userClusterConfig.Clusters` it's a slice, I know this is only for the python tests and the generated config file that glues the configuration but is not consistent with the design.

Reply:
Ah sorry I see what you are saying. Because the python tests need a standard user cluster, and the go tests need a single admin cluster. Isn't that the case?

Reply:
I get it but if we need multiple clusters for go tests what's the steps to follow.

Should we proceed with something like the `userClusterConfig` struct instead of using the `rancher.Config` and create a separate file?

Reply:
I guess the boils down to changing the current design. Currently the framework only supports the single cluster in the config file when running validations, if we keep this design then yes in a Jenkins environment I think we would still need the userClusterConfig to update the CATTLE_CONFIG_TEST for multiple jobs.

Reply:
I think I am facing this issue with the registries tests. 

I have two clusters besides the local to run the same go tests with different values. I'll try to follow the suggested `userClusterConfig` approach.

Reply:
I think that is a good idea, but also would it make sense to update the framework to support multiple clusters?